pygfx/renderers/wgpu/lineshader.py

```diff
@@ -311,138 +333,453 @@ def code_vertex_core(self):
+            }
+            let opacity = min(1.0, color.a) * alpha * u_material.opacity;
+
+            //let opacity_multiplier = select(-1.0, 1.0, !is_front);
+            let out_color = vec4<f32>(physical_color, opacity);
+
+            // Wrap up
+            apply_clipping_planes(varyings.world_pos);
+            var out = get_fragment_output(varyings.position.z, out_color);
+
+            // Set picking info.
+            $$ if write_pick
+            // The wobject-id must be 20 bits. In total it must not exceed 64 bits.
+            // The pick_idx is int-truncated, so going from a to b, it still has the value of a
+            // even right up to b. The pick_zigzag alternates between 0 (even indices) and 1 (odd indices).
+            // Here we decode that. The result is that we can support vertex indices of ~32 bits if we want.
+            let is_even = varyings.pick_idx % 2u == 0u;
+            var coord = select(varyings.pick_zigzag, 1.0 - varyings.pick_zigzag, is_even);
+            coord = select(coord, coord - 1.0, coord > 0.5);
+            let idx = varyings.pick_idx + select(0u, 1u, coord < 0.0);
+            out.pick = (
+                pick_pack(u32(u_wobject.id), 20) +
+                pick_pack(u32(idx), 26) +
+                pick_pack(u32(coord * 100000.0 + 100000.0), 18)
+            );
+            $$ endif
+
+            // The outer edges with lower alpha for aa are pushed a bit back to avoid artifacts.
+            // This is only necessary for blend method "ordered1"
+            //out.depth = varyings.position.z + 0.0001 * (0.8 - min(0.8, alpha));
+
+            return out;
+        }
+        """
+
+
+@register_wgpu_render_function(Line, LineDashedMaterial)
+class LineDashedShader(LineShader):
+    needs_bake_function = True
+
+    def __init__(self, wobject):
+        super().__init__(wobject)
+
+        n_verts = wobject.geometry.positions.nitems
+        distance_array = np.zeros((n_verts,), np.float32)
+        self.line_distance_buffer = Buffer(distance_array)
+        self._positions_hash = None
+
+    def _get_extra_bindings(self, wobject):
+        rbuffer = "buffer/read_only_storage"
+        return [
+            Binding("s_cumdist", rbuffer, self.line_distance_buffer, "VERTEX"),
+        ]
+
+    def bake_function(self, wobject, camera, logical_size):
+        # Prepare
+        positions_buffer = wobject.geometry.positions
+        dash_offset = wobject.material.dash_offset
+        r_offset, r_size = positions_buffer.draw_range
+
+        # Prepare arrays
+        positions_array = positions_buffer.data[r_offset : r_offset + r_size]
+        distance_array = self.line_distance_buffer.data[r_offset : r_offset + r_size]
+
+        # Get vertices in the appropriate coordinate frame
+        if wobject.material.dash_is_screen_space:
+            xyz = la.vec_transform(positions_array, camera.camera_matrix)
+            vertex_array = xyz[:, :2] * 0.5 * np.array(logical_size)
```

Comment:
minor optimization thing:

```suggestion
            vertex_array = xyz[:, :2] * (0.5 * np.array(logical_size))
```